jax_unirep/evotuning.py

```diff
@@ -314,37 +352,41 @@ def step(i, state):
     epoch_len = round(len(sequences) / batch_size)
 
     n = n_epochs * epoch_len
-    for i in range(n):
+    for i in tqdm(range(n), desc="Iteration"):
 
         if i % epoch_len == 0:
             logger.info(f"Starting epoch {int(i / epoch_len) + 1}")
-
-        logging.debug("Getting batches")
+        logger.debug(f"Iteration {i}")
+        logger.debug("Getting batches")
         l = choice(seq_lens)
         x, y = len_batching_funcs[l]()
 
         # actual forward & backwrd pass happens here
-        logging.debug("Getting state")
-        state = step(i, state)
+        logger.debug("Getting state")
+        state = step(i, x, y, state)
 
         if i % epoch_len == 0:
```

Comment:
I think this whole conditional is executed in vain if `loss` is not being used further down in the `steps_per_print` conditional. I guess we should either print the loss here in any case, or move the whole thing down into the `steps_per_print` conditional?

Reply:
Yes, this piece is a bit messy. I think it can be simplified. Let me push something up.